versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/IndexesLogicImpl.java

```diff
@@ -537,26 +539,38 @@ void completeIndexesInCommitChain(
 
       StoreIndex<CommitOp> newIndex;
       ObjId referenceIndex;
+      List<IndexStripe> indexStripes;
       if (parent != null) {
         newIndex = incrementalIndexForUpdate(parent, Optional.empty());
         referenceIndex = parent.referenceIndex();
+        indexStripes = parent.referenceIndexStripes();
       } else {
         newIndex = newStoreIndex(COMMIT_OP_SERIALIZER);
         referenceIndex = null;
+        indexStripes = Collections.emptyList();
       }
 
       commitOperations(current).forEach(newIndex::add);
 
-      CommitObj newCommit =
+      CommitObj.Builder c =
           commitBuilder()
               .from(current)
               .incompleteIndex(false)
               .referenceIndex(referenceIndex)
-              .incrementalIndex(newIndex.serialize())
-              .build();
-      newCommit = commitLogic.updateCommit(newCommit);
+              .referenceIndexStripes(indexStripes)
+              .incrementalIndex(newIndex.serialize());
+
+      if (parent != null) {
+        int parents = Math.min(parentsPerCommit - 1, parent.tail().size());
+        List<ObjId> tail = new ArrayList<>(parents + 1);
+        tail.add(parent.id());
+        for (int j = 0; j < parents; j++) {
+          tail.add(parent.tail().get(j));
```

Comment:
nit: I wonder if `.addAll()` with a sub list might be more readable (if not more performant)?